lineapy/graph_reader/program_slice.py

```diff
@@ -262,27 +269,78 @@ def get_source_code_from_graph(program: Graph) -> CodeSlice:
                 )
             )
 
+    # print(slice_nodes)
+    # for n in slice_nodes:
+    #     nn = session_graph.ids[n]
+    #     print(n, nn.node_type)
+    #     if hasattr(nn, "name"):
+    #         print(nn.name)
+    for n in session_graph.nodes:
+        # print(n.id)
+        if n.source_location is None:
+            # print("no source", n.id, n.node_type)
+            continue
+        # ignore the code cells that have no part in the slice at all
```

Comment:
Followup, is this necessary for correctness or just for speeding up the case for `not include_non_slice_as_comment` ? 
The iteration in L296 has 
`lines = source_code_to_lines.get(source_code, set())`  
and source_code_to_lines comes from the slice, seeming to indicate this might still work without this line here. 

If this is an optimization can we clarify it as such?

Reply:
having source location not empty is critical for get_code_from_graph. This piece does not deal with code generation at all (most of code generation happens in artifact_collection I think). So this is closer to "required for correctness/ not crashing" than an optimization.

Reply:
gotcha, cool